src/GoTrueClient.ts

```diff
@@ -855,6 +900,36 @@ export default class GoTrueClient {
     }
   }
 
+  private async _waitOnRefreshTokenSync(): Promise<void> {
+    return new Promise((resolve) => {
+      this.refreshTokenSyncIntervalTimer = setInterval(async () => {
+        const refreshTokenSync = (await getItemAsync(
+          this.storage,
+          this.refreshTokenSyncStorageKey
+        )) as SyncTokenRefreshStorage | undefined
+
+        if (refreshTokenSync?.status === 'TOKEN_REFRESHED') {
+          clearInterval(this.refreshTokenSyncIntervalTimer)
+          clearTimeout(this.refreshTokenSyncTimeoutTimer)
+          resolve()
+        }
+      }, 100)
+
+      // Stop interval if tokenSync.status does not change
+      this.refreshTokenSyncTimeoutTimer = setTimeout(async () => {
+        await removeItemAsync(this.storage, this.refreshTokenSyncStorageKey)
+        clearInterval(this.refreshTokenSyncIntervalTimer)
+        resolve()
+      }, 1500)
+    })
+  }
```

Comment:
I saw there's an exponential backoff for network retry. 
I didn't think it was required here. A request to refresh a token takes on average 150-300ms.